### ISSUE TITLE
switching git branches resets custom icons

### ISSUE THREAD

**Issue by YehudaK** [Unity-Technologies/NotificationsSamples#10]
Every time I switch git branches, all custom icons set in **Edit->Project Settings->Mobile Notification Settings** gets reset and I have to create icons again. Is there any specific file in Unity project which I should track to prevent this?

Thanks

**Comment by YehudaK** [comment]
The problem is that all changes are successfully saved in /Assets/Editor/com.unity.mobile.notifications/NotificationSettings.asset and later merged to release branch BUT once I start Unity, it removes those lines bellow:

```
  AndroidNotificationEditorSettingsValues:
    keys:
    - UnityNotificationAndroidRescheduleOnDeviceRestart
    values:
    - False
  TrackedResourceAssets:
  - Id: icon_fully_recovered
    Type: 0
    Asset: {fileID: 2800000, guid: 7854c247df753814fbab6838b04909ac, type: 3}
  - Id: icon_continue_play
    Type: 0
    Asset: {fileID: 2800000, guid: b4d30efc27d2f45449a5d936773e0573, type: 3}
  - Id: icon_app
    Type: 1
    Asset: {fileID: 2800000, guid: 3615aca4454f2454e8d35194b8b9a7aa, type: 3}
    keys: []
    values: []
```

and replaces them with this:

```
  AndroidNotificationEditorSettingsValues:
    keys: []
    values: []
  TrackedResourceAssets: []
```

**Comment by ajon542** [comment]
Experiencing the same thing.
Not sure if it matters but we're on Unity 2018.4.2f1

**Comment by ajon542** [comment]
It looks like in the Initialize method for the UnityNotificationEditorManager class (in the mobile notifications package), the AssetDatabase.LoadAssetAtPath is failing if the "Library" directory is deleted.

**Comment by YehudaK** [comment]
@ajon542 do you know where the sources for Notification package is located and can we modify them like plugins? We can try to debug and fix the issue. It's really annoying to re-add icons before every update release.

**Comment by ajon542** [comment]
It appears that 1.0.3-preview package has been released with a fix for this issue:
https://forum.unity.com/threads/mobile-notification-preview-package.620227/page-4#post-4777439

To answer your question directly though, the source appears to live in the "Library/PackageCache" directory. I actually removed mobile notifications from the package manager and pulled the source into the "Assets" directory so I could tinker.

**Comment by YehudaK** [comment]
Thanks @ajon542 I tried but it did not work (after restart settings gets reset), also they increase min Android version to 4.4 which is never welcome as I do not like to lose 5% of users because of this plugin =)

I already bought **[Simple Android Notifications](https://assetstore.unity.com/packages/tools/integration/simple-android-notifications-71294)** plugin but still never had a chance to give it a try. I should also try to pull the source and modify it as you did, thanks for your tip!

**SOLUTION: simply place icons in /Assets/Plugins/Android/res/drawable and that's it. I tested and it's working with small/large icons. Later you can access them with their name without png extension =)**

**Comment by ajon542** [comment]
Ohh no, you are right. I just read the "fixes" section and assumed. I just tried and NotificationSettings are reset with 1.0.3-preview as well :(

**Comment by qwyt** [comment]
This is fixed in 1.0.3-preview.6, the issue with reseting settings fixed in an earlier version was a caused by different reasons.
